Sources/Fullscreen/Explore/ExploreView.swift

```diff
@@ -37,23 +58,24 @@ public final class ExploreView: UIView {
 
     public weak var delegate: ExploreViewDelegate?
     public weak var dataSource: ExploreViewDataSource?
+    public weak var recommendationsDataSource: ExploreViewRecommendationsDatasource?
```

Comment:
This should not be `public` since the registration of cells happens in the init, aka. before this value would be set if you're creating the view like this:

```swift
let view = ExploreView(withAutoLayout: true)	// DataSource is being asked for cells to display in setup().
view.recommendationsDataSource = someClass      // DataSource is set.
```

The reason it's working in the app right now is because the cell `StandardAdRecommendationCell` is already being registered when creating the `collectionView` (line 91), but this will break if you supply other cells or remove the current registration :)

Reply:
Yes, you actually have to use the init that sets recommendationDataSource, so I should remove the option to init without and make this private. Good catch.